RecoLocalCalo/HGCalRecAlgos/src/HGCal3DClustering.cc

```diff
@@ -102,9 +102,13 @@ std::vector<reco::HGCalMultiCluster> HGCal3DClustering::makeClusters(const reco:
 
       }
       if( temp.size() > minClusters ) {
+	math::XYZPoint position = clusterTools->getMultiClusterPosition(temp);
+	if (abs(position.z()) < 100) continue;
```

Comment:
Well, it depends on what criterion we really want. If we just want to exclude the (0.,0.,0.) default value, it would probably be better to compare the values to 0. (usually float comparisons are a bad idea, but in this case it should be okay). But if we really want to drop any multicluster with z outside of HGCal, the actual HGCal geometry should be used to determine the z boundary. (Otherwise, the code will have maintenance issues and inconsistencies, as we've already seen with other hardcoded geometry numbers.)